quilt/tools/command.py

```diff
@@ -410,14 +410,23 @@ def _build_internal(package, path, dry_run, env):
             try:
                 _clone_git_repo(url, branch, tmpdir)
                 build_from_path(package, tmpdir, dry_run=dry_run, env=env)
+            except yaml.scanner.ScannerError as ex:
+                message_parts = str(ex).split('\n')
+                message_parts.insert(0, "Syntax error while reading {!r}".format(path))
+                raise CommandException('\n  '.join(message_parts))
             except Exception as exc:
                 msg = "attempting git clone raised exception: {exc}"
                 raise CommandException(msg.format(exc=exc))
             finally:
                 if os.path.exists(tmpdir):
                     rmtree(tmpdir)
         else:
-            build_from_path(package, path, dry_run=dry_run, env=env)
+            try:
+                build_from_path(package, path, dry_run=dry_run, env=env)
+            except yaml.scanner.ScannerError as ex:
+                message_parts = str(ex).split('\n')
+                message_parts.insert(0, "Syntax error while reading {!r}".format(path))
+                raise CommandException('\n  '.join(message_parts))
```

Comment:
duplicate code - can you move into build_from_path() ?

Reply:
I could, however, move that section to a function, and try: except: that, or try:except the whole block, or send the path into build_from_path as a new param "original_path" or some such..

I didn't know if the small amount of duplication was worth shuffling things about for, but I'd be happy to find a way.

Reply:
ah, I mean move the except block down into build_from_path:

def build_from_path(package, path, dry_run=False, env='default'):
...

    try:
...
   except yaml.scanner.ScannerError as ex:
        message_parts = str(ex).split('\n')
        message_parts.insert(0, "Syntax error while reading {!r}".format(path))
        raise CommandException('\n  '.join(message_parts))
    except BuildException as ex:
        raise CommandException("Failed to build the package: %s" % ex)

Reply:
Ah, I missed the ability to compile from a directory, and thought that 'build from path' took a specific YML file (only).  I see what it's doing now, and I'll shift that over.

Reply:
I had originally had it there so it would catch the original `path` during git repo pulls, instead of the temp dir sent to build_from_path.